docs/glossary.md

```diff
@@ -31,3 +31,19 @@
   - Defending units can only participate in this phase if `Submarines Defending May Submerge Or Retreat` property is set.
   - `Sub Retreat Before Battle` defines if this phase occurs before the First Strike Phase or at the end of the Battle Round.
 - **Submerge Subs Phase**: Another name for **Evasion Phase** and generally used when only submarines have `canEvade` and `Submersible Subs` is set.
+
+##Development Terms
+- **Branch**: A Branch is a copy of the main codebase.  It is used to keep changes separate until they are completed or in a stable state.  
```

Comment:
- A branch is an ordered set of commits.
- A feature branch is a specific type of branch we use in the release-branch workflow.
- A release branch is another type of branch we use to make patches to released versions of code easier to deploy.
- Completed and stable should generally be synonymous. If they are not, and a branch can be completed and merged, it implies it is up to the maintainers to create stability and that we are pushing unstable and buggy code. It is the old SVN big-bang-merge-before-release workflow that classically used an unstable/stable model. In that model there would be no changes to the master branch for months at a time. Everyone would get a copy of the code, make changes, and then email patches into. The maintainer would take all patches, merge them altogether and create the 'unstable' release. Because none of these changes were tested with one another, or perhaps at all, this release was unstable. The main codebase was then frozen while bug fixes were done to get it to stable and that was then a second, stable release. This process was then repeated ad-infinitum.

Now instead we strive for changes that are small and complete unto themselves and well tested. It's been a pain point where the maintainers have to spend a few weeks before a release finding and rooting out bugs that would have been better never introduced, either through more testing and often particularly by more seriously adding automated testing.